src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -263,85 +262,101 @@ public IEnumerable<PackageItemViewModel> GetCurrent()
             }
 
-            var listItemViewModels = new List<PackageItemViewModel>(capacity: _state.ItemsCount);
+            var listItemViewModels = new Dictionary<string, PackageItemViewModel>();
 
             foreach (PackageSearchMetadataContextInfo metadataContextInfo in _state.Results.PackageSearchItems)
             {
-                VersionRange allowedVersions = VersionRange.All;
-                VersionRange versionOverride = null;
+                var packageId = metadataContextInfo.Identity.Id;
+                var packageVersion = metadataContextInfo.Identity.Version;
 
-                // get the allowed version range and pass it to package item view model to choose the latest version based on that
-                if (_packageReferences != null)
+                if (listItemViewModels.TryGetValue(packageId, out PackageItemViewModel exisitngListItem))
                 {
-                    IEnumerable<IPackageReferenceContextInfo> matchedPackageReferences = _packageReferences.Where(r => StringComparer.OrdinalIgnoreCase.Equals(r.Identity.Id, metadataContextInfo.Identity.Id));
-                    VersionRange[] allowedVersionsRange = matchedPackageReferences.Select(r => r.AllowedVersions).Where(v => v != null).ToArray();
-                    VersionRange[] versionOverrides = matchedPackageReferences.Select(r => r.VersionOverride).Where(v => v != null).ToArray();
+                    exisitngListItem.InstalledVersions = exisitngListItem.InstalledVersions.Append(packageVersion);
+                }
+                else
+                {
+                    VersionRange allowedVersions = VersionRange.All;
+                    VersionRange versionOverride = null;
 
-                    if (allowedVersionsRange.Length > 0)
+                    // get the allowed version range and pass it to package item view model to choose the latest version based on that
+                    if (_packageReferences != null)
                     {
-                        allowedVersions = allowedVersionsRange[0];
+                        IEnumerable<IPackageReferenceContextInfo> matchedPackageReferences = _packageReferences.Where(r => StringComparer.OrdinalIgnoreCase.Equals(r.Identity.Id, metadataContextInfo.Identity.Id));
+                        VersionRange[] allowedVersionsRange = matchedPackageReferences.Select(r => r.AllowedVersions).Where(v => v != null).ToArray();
+                        VersionRange[] versionOverrides = matchedPackageReferences.Select(r => r.VersionOverride).Where(v => v != null).ToArray();
+
+                        if (allowedVersionsRange.Length > 0)
+                        {
+                            allowedVersions = allowedVersionsRange[0];
+                        }
+
+                        if (versionOverrides.Length > 0)
+                        {
+                            versionOverride = versionOverrides[0];
+                        }
```

Comment:
We can use FirstOrDefault to make this code more concise. (Please verify correctness):
```suggestion
                        allowedVersions = allowedVersionsRange.FirstOrDefault() ?? VersionRange.All;                            
                        versionOverride = versionOverrides.FirstOrDefault();
```